README.md

```diff
@@ -136,6 +137,78 @@ $server->on('error', function (Exception $e) {
 Note that the request object can also emit an error.
 Check out [request](#request) for more details.
 
+### Middleware
+
+Middleware can ba added to the server using [`MiddlewareRunner`](src/MiddlewareRunner.php)
+instead of the the `callable`. A middleware is expected adjure to the following rules:
+
+* It is a `callable`.
+* It accepts `ServerRequestInterface` as first argument and `callable` as second argument.
+* It either returns a promise resolving to a `ResponseInterface`; or it calls the second argument
+with a `ServerRequestInterface`.
+
+The following example adds a middleware that adds the current time to the request as a 
+header (`Request-Time`) and middleware that always returns a 200 code without a body: 
+
+```php
+$server = new Server(new MiddlewareRunner(
+    new Response(404), // Default response if no middleware returns a promise
+    [
+        function (ServerRequestInterface $request, callable $next) {
+            $request = $request->withHeader('Request-Time', time());
+            return $next($request);
+        }
+        function (ServerRequestInterface $request, callable $next) {
+            return new Response(200);
+        }
+    ]
+));
+```
+
+#### Buffer
```

Comment:
The `Buffer` and `LimitHandlers` are not part of this PR and the sections should probably removed.

Reply:
Yes correct, a PR for `Buffer` is coming soon and `LimitHandlers` after that. Then two more PR's will follow, one for the readme section and one for examples